domain/src/main/java/com/teamagam/gimelgimel/domain/messages/poller/PolledMessagesProcessor.java

```diff
@@ -40,132 +35,124 @@
-    mAddPolledMessageToRepositoryInteractorFactory = addPolledMessageToRepositoryInteractorFactory;
+    mAddMessageToRepositoryInteractorFactory = addMessageToRepositoryInteractorFactory;
+    mAddAlertRepositoryInteractorFactory = addAlertRepositoryInteractorFactory;
     mProcessNewVectorLayerInteractorFactory = processNewVectorLayerInteractorFactory;
-    mProcessIncomingAlertMessageInteractorFactory = processIncomingAlertMessageInteractorFactory;
-    mMessageProcessorVisitor = new MessageProcessorVisitor();
   }
 
   @Override
-  public void process(Collection<Message> polledMessages) {
+  public void process(ChatMessage polledMessage) {
+    if (polledMessage == null) {
+      throw new IllegalArgumentException("polledMessages cannot be null");
+    }
+
+    processMessage(polledMessage);
+  }
+
+  @Override
+  public void process(Collection<ChatMessage> polledMessages) {
     if (polledMessages == null) {
       throw new IllegalArgumentException("polledMessages cannot be null");
     }
 
     sLogger.d("MessagePolling service processing " + polledMessages.size() + " new messages");
 
-    for (Message msg : polledMessages) {
+    for (ChatMessage msg : polledMessages) {
       processMessage(msg);
     }
   }
 
-  private void processMessage(Message message) {
+  @Override
+  public void process(VectorLayer vectorLayer) {
+    mProcessNewVectorLayerInteractorFactory.create(vectorLayer).execute();
+  }
+
+  @Override
+  public void process(UserLocation userLocation) {
+    if (!mPreferencesUtils.isMessageFromSelf(userLocation.getUser())) {
+      mUsersLocationRepository.add(userLocation);
+    }
+  }
+
+  private void processMessage(ChatMessage message) {
     if (message == null) {
       throw new IllegalArgumentException("Message cannot be null");
     }
 
-    message.accept(mMessageProcessorVisitor);
+    MessageProcessorVisitor visitor = new MessageProcessorVisitor(message);
+    message.accept(visitor);
   }
 
-  private class MessageProcessorVisitor implements IMessageVisitor {
+  private class MessageProcessorVisitor implements IMessageFeatureVisitor {
 
-    @Override
-    public void visit(MessageGeo message) {
-      addToMessagesRepository(message);
-      mapEntityToMessage(message, message.getGeoEntity());
-      displayGeoEntity(message.getGeoEntity());
-    }
+    private ChatMessage mMessage;
 
-    @Override
-    public void visit(MessageText message) {
-      addToMessagesRepository(message);
-    }
+    public MessageProcessorVisitor(ChatMessage message) {
+      mMessage = message;
 
-    @Override
-    public void visit(MessageImage message) {
-      addToMessagesRepository(message);
-      if (message instanceof MessageGeoImage) {
-        MessageGeoImage messageGeoImage = (MessageGeoImage) message;
-        mapEntityToMessage(messageGeoImage, messageGeoImage.getGeoEntity());
-        displayGeoEntity(messageGeoImage.getGeoEntity());
-      }
+      addToMessagesRepository(mMessage);
```

Comment:
My mistake. It's a proper message. The visitor is only an opporunity to deal with the different features of it.